### ISSUE TITLE
[Bug]: LuSolvePrepared VJP silently drops cotangent for matrix A (always returns None)

### ISSUE THREAD

**Issue by terasakisatoshi** [tensor4all/tenferro-rs#1151]
### What happened?

In `crates/tenferro-linalg/src/ad/rules/solve.rs:488-509`, the `transpose_lu_solve_prepared` VJP rule only computes the cotangent for `active_mask[3]` (the RHS `b`). It returns `vec![None, None, None, None]` with `None` at indices 0, 1, and 2 regardless of whether those inputs are active in the mask.

```rust
let mut result = vec![None, None, None, None];
if active_mask[3] {
    // ... only handles cotangent for input index 3 (b)
    result[3] = Some(out[0]);
}
result  // indices 0, 1, 2 always None
```

This means:
- **The cotangent for matrix A (index 0) is silently dropped**
- The packed LU factor (index 1) and pivots (index 2) cotangents are also dropped

Automatic differentiation through `lu_solve_prepared` **cannot backpropagate gradients to the matrix A**. If a user needs `∂f/∂A` where `f` is a function involving a prepared LU solve, the gradient is incorrectly zero.

Note also that this function returns `Vec<Option<LocalValueId>>` directly (not `ADRuleResult`), so errors cannot be propagated even if the lack of support for indices 0-2 is considered an error condition.

### What did you expect?

If differentiating through the LU factor/pivots is intentionally unsupported, the rule should return an error for active indices 0-2 rather than silently returning `None`. If support is planned, the VJP formula needs to be implemented.

### Minimal reproducer

```rust
// Use lu_solve_prepared in an AD computation with requires_grad on the input matrix A.
// The gradient for A will be None/zero regardless of the cotangent from downstream.
```

### Affected area

- Automatic differentiation
- Linear algebra (solve/lu)

### Backend/device

- CPU BLAS/LAPACK backend

### tenferro-rs version or commit

`main` at d911541b

### Environment

Any system using `tenferro-linalg` AD with `lu_solve_prepared`.

### Verification hint

Check whether `transpose_lu_solve_prepared` should return an error for active indices 0-2 or if the VJP formula for the LU factor is too expensive to implement. If the latter, document the limitation.

**Comment by shinaoka** [comment]
Resolved by merged PR #1203.
